specification/schemas/BaseShipment.json

```diff
@@ -0,0 +1,197 @@
+{
+  "type": "object",
+  "allOf": [
+    {
+      "$ref": "#/components/schemas/MinimalShipment"
+    },
+    {
+      "properties": {
+        "attributes": {
+          "type": "object",
+          "properties": {
+            "recipient_address": {
+              "$ref": "#/components/schemas/BaseAddress"
+            },
+            "return_address": {
+              "$ref": "#/components/schemas/BaseAddress"
+            },
+            "sender_address": {
+              "$ref": "#/components/schemas/BaseAddress"
+            },
+            "pickup_location": {
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "code": {
+                  "type": "string",
+                  "example": "205604"
+                },
+                "address": {
+                  "$ref": "#/components/schemas/BaseAddress"
+                }
+              }
+            },
+            "description": {
+              "type": [
+                "string",
+                "null"
+              ],
+              "example": "Order #8008135"
+            },
+            "insurance": {
+              "oneOf": [
+                {
+                  "$ref": "#/components/schemas/BasePrice"
+                },
+                {
+                  "type": "null"
+                }
+              ]
+            },
+            "physical_properties": {
+              "$ref": "#/components/schemas/BasePhysicalProperties"
+            },
+            "items": {
+              "oneOf": [
+                {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/BaseShipmentItem"
+                  }
+                },
+                {
+                  "type": "null"
+                }
+              ]
+            },
+            "register_at": {
+              "oneOf": [
+                {
+                  "allOf": [
+                    {
+                      "$ref": "#/components/schemas/Timestamp"
+                    },
+                    {
+                      "readOnly": false,
```

Comment:
Why is there a `"readOnly": false` here?

Reply:
The Timestamp schema has `"readOnly": true` inside it, because of the `created_at` and `updated_at` timestamp values being `readOnly`. I added `"readOnly": false` here, instead of adding `"readOnly": true` to the other times the `Timestamp` schema is used because it's more common for a `Timestamp` to be `readOnly` than not.

Reply:
But won't this break when I try to post a shipment with `register_at` set? I don't know if readonly is enforced anywhere, but if the shipment is to comply with **all*Of these schema's it will fail at the `register_at` not being readonly.

Reply:
readOnly is not enforced. I just tried it, and even with readOnly true it still lets it through... In the ui though, it shows readOnly false instead of true.